types/node/index.d.ts

```diff
@@ -2828,13 +2828,17 @@ declare module "fs" {
      * @param encoding
      */
     export function readFileSync(filename: string, encoding: string): string;
+    export function readFileSync(filename: string, encoding: null): Buffer;
```

Comment:
As tsconfig.json for node has strictNullChecks set to false the overload with encoding: null needs to be first as the encoding: string variant matches also null.